src/openms/source/SYSTEM/File.cpp

```diff
@@ -116,6 +117,23 @@ namespace OpenMS
     return !fi.exists() || fi.size() == 0;
   }
 
+  bool File::rename(const String& from, const String& to, bool overwrite_existing, bool verbose)
+  {
+    // existing file? Qt won't overwrite, so try to remove it:
+    if (overwrite_existing && exists(to) && !remove(to))
```

Comment:
I think you should check if from == to

Reply:
mhh... that opens quite a can of worms...
It could be relative filenames, symlinks etc... which point to the same file. Checking `from == to` is only to tip of the iceberg

Reply:
http://doc.qt.io/archives/qt-4.8/qfileinfo.html#canonicalFilePath could solve it...

Reply:
hmm otherwise you might easily remove a file that you just wanted to rename...